### PR TITLE
Support PostgreSQL monitoring. 

### LINKED ISSUE #9472: [Feature] Monitoring PostgreSQL Server
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Right now, we have a `MySQL` Monitoring, and I want to add the `PostgreSQL` Server Monitoring to enhance SkyWalking's database monitoring.
The dataflow would be `PostgreSQL Server` -> `postgres_exporter` -> `OpenTelemetry Collector` -> `SkyWalking OpenCensus Receiver` -> `MAL engine`.

### Steps
1. Discuss which metrics we should do for the `PostgreSQL` server.
2. Add `PostgreSQL` layer to OAP, and begin to set up dashboards.
3. Add `PostgreSQL` as a sub of the `Database` layer.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yswdqz**
### <Feature description>
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #9472 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

Screenshots:
image
image
image

### CODE REVIEW

**Comment by yswdqz** [comment]
I found that if I add the license to `mock.sql` , it will fail.
image

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:132]: Don't put a subjective word(powerful) in the comment when refer to 3rd party project.
Such as in this case, ppl could think we only accept monitoring something powerful as a layer. This is not our meaning. 😁

**Comment by wu-sheng** [comment]
> I found that if I add the license to `mock.sql` , it will fail.
> 
> image
> 
> 
> 
> This is probably because of PostgreSQL, when I do this by MySQL , It will success.

You could check the PostgreSQL doc, which style comment do they support, such as -- or something else.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java
Comment (yswdqz) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:132]: I just copy from its official website. I don't know how to describe it.

**Comment by kezhenxu94** [comment]
> I found that if I add the license to `mock.sql` , it will fail. image
> 
> This is probably because of PostgreSQL, when I do this by MySQL , It will success.

The standard comment style of SQL is `--`, mysql allows `#` maybe just because mysql has extended syntax for that

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java
Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:132]: I like this version, it seems more objective.

> PostgreSQL is an advanced, enterprise-class, and open-source relational database system. 

Found from https://www.postgresqltutorial.com/postgresql-getting-started/what-is-postgresql/

**Comment by wu-sheng** [comment]
BTW, do you have env to work on showcase now? Or still need extra setup from us?

**Comment by yswdqz** [comment]
> BTW, do you have env to work on showcase now? Or still need extra setup from us?

 Sorry. My environment still doesn't work.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java
Comment (yswdqz) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:132]: Got it.

**Comment by wu-sheng** [comment]
> > BTW, do you have env to work on showcase now? Or still need extra setup from us?
> 
> 
> 
>  Sorry. My environment still doesn't work.

FYI @kezhenxu94 This would need your helps again after merging.

**Comment by wu-sheng** [comment]
@yswdqz You could sync the master branch and update accordingly.

**Comment by wu-sheng** [comment]
In #9497, we have provided `enabledOtlpRules`, after you set `postgre` at `oc` and `oltp` rules, could you use `oltp` exporter in OTEL collector to verify the results whether they are as same as the current setup.

**Comment by yswdqz** [comment]
I get it.

**Comment by yswdqz** [comment]
I have a problem when I use the oltp exporter in OTEL collector: The OAP can't receive any data.
The log of the collector is following:
```
{"level":"error","ts":"2022-08-22T00:41:08.619+0800","caller":"exporterhelper/queued_retry.go:149","msg":"Exporting failed. Try enabling retry_on_failure config option to retry on retryable errors","kind":"exporter","data_type":"metrics","name":"opencensus","error":"EOF","stacktrace":"go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/queued_retry.go:149\ngo.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/metrics.go:132\ngo.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).send\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/queued_retry.go:83\ngo.opentelemetry.io/collector/exporter/exporterhelper.NewMetricsExporter.func2\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/metrics.go:112\ngo.opentelemetry.io/collector/consumer.ConsumeMetricsFunc.ConsumeMetrics\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/consumer/metrics.go:36\ngo.opentelemetry.io/collector/service/internal/fanoutconsumer.(*metricsConsumer).ConsumeMetrics\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/service/internal/fanoutconsumer/metrics.go:75\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:297\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:176\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:143"}
{"level":"error","ts":"2022-08-22T00:41:08.619+0800","caller":"exporterhelper/queued_retry.go:85","msg":"Exporting failed. Dropping data. Try enabling sending_queue to survive temporary failures.","kind":"exporter","data_type":"metrics","name":"opencensus","dropped_items":495,"stacktrace":"go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).send\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/queued_retry.go:85\ngo.opentelemetry.io/collector/exporter/exporterhelper.NewMetricsExporter.func2\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/exporter/exporterhelper/metrics.go:112\ngo.opentelemetry.io/collector/consumer.ConsumeMetricsFunc.ConsumeMetrics\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/consumer/metrics.go:36\ngo.opentelemetry.io/collector/service/internal/fanoutconsumer.(*metricsConsumer).ConsumeMetrics\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/service/internal/fanoutconsumer/metrics.go:75\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:297\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:176\ngo.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle\n\tC:/Users/runneradmin/go/pkg/mod/go.opentelemetry.io/collector@v0.55.0/processor/batchprocessor/batch_processor.go:143"}
{"level":"warn","ts":"2022-08-22T00:41:08.619+0800","caller":"batchprocessor/batch_processor.go:178","msg":"Sender failed","kind":"processor","name":"batch","pipeline":"metrics","error":"EOF"}
```
This log might be wrong because I have not use batch processor and anything about opencensus. But when I change exporter and OAP to opencensus , this log will not appear, and I can found the data.
Followings are my OAP configs and collector config.
```
receiver-otel:
  selector: ${SW_OTEL_RECEIVER:default}
  default:
    enabledHandlers: ${SW_OTEL_RECEIVER_ENABLED_HANDLERS:"otlp"}
    enabledOcRules: ${SW_OTEL_RECEIVER_ENABLED_OC_RULES:"istio-controlplane,k8s-node,oap,vm,mysql,postgresql"}
    enabledOtlpRules: ${SW_OTEL_RECEIVER_ENABLED_OTLP_RULES:"istio-controlplane,k8s-node,oap,vm,mysql,postgresql"} 
```
```
receivers:
  prometheus:
    config: 
     scrape_configs:
       - job_name: 'postgresql-monitoring'
         scrape_interval: 10s
         static_configs:
           - targets: ['postgres-service:9187']
           
processors:

exporters:
  otlp:
    endpoint: oap:11800
    tls:
      insecure: true
  logging:
    loglevel: debug
service:
  pipelines:
    metrics:
      receivers:
      - prometheus
      processors: []
      exporters:
      - logging
      - otlp
```

**Comment by wu-sheng** [comment]
@kezhenxu94 Could you check this?

**Comment by wu-sheng** [comment]
@yswdqz Could you resolve the conflicts and keep all things in OC format for now?

Zhenxu is testing on showcase(https://github.com/apache/skywalking-showcase/tree/otlp) to see whether otlp could replace all oc, or whether there are some bugs.

**Comment by wu-sheng** [comment]
@yswdqz Do you enable the handler?

```yaml
receiver-otel:
  selector: ${SW_OTEL_RECEIVER:default}
  default:
    enabledHandlers: ${SW_OTEL_RECEIVER_ENABLED_HANDLERS:"oc,otlp"}
```

@kezhenxu94 Should we enable both in default? I noticed you use System env to override `oc` by `otlp` in your PR. Could we have both? I think it is much easier. Further, should we merge `enabledOcRules` and ` enabledOtlpRules`?

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:362]: Suggest adding a unit or specific explanation for the `Trend`, same as the other place.

**Review by wankai123 — DISMISSED** [review]

**Comment by yswdqz** [comment]
> @yswdqz Do you enable the handler?
> 
> ```yaml
> receiver-otel:
>   selector: ${SW_OTEL_RECEIVER:default}
>   default:
>     enabledHandlers: ${SW_OTEL_RECEIVER_ENABLED_HANDLERS:"oc,otlp"}
> ```
> 
> @kezhenxu94 Should we enable both in default? I noticed you use System env to override `oc` by `otlp` in your PR. Could we have both? I think it is much easier. Further, should we merge `enabledOcRules` and ` enabledOtlpRules`?

Yes, I have resolved the conflicts and keep all things in OC format? And the OAP logs can confirm that I enable it.
```
:35:59,783 - org.apache.skywalking.oap.server.starter.config.ApplicationConfigLoader - 69 [main] INFO  [] - Get a provider define belong to receiver-otel module, provider name: default
2022-08-22 00:35:59,783 - org.apache.skywalking.oap.server.starter.config.ApplicationConfigLoader - 118 [main] INFO  [] - Provider=default config=enabledHandlers has been set as otlp
2022-08-22 00:35:59,783 - org.apache.skywalking.oap.server.starter.config.ApplicationConfigLoader - 118 [main] INFO  [] - Provider=default config=enabledOcRules has been set as istio-controlplane,k8s-node,oap,vm,mysql,postgresql
2022-08-22 00:35:59,783 - org.apache.skywalking.oap.server.starter.config.ApplicationConfigLoader - 118 [main] INFO  [] - Provider=default config=enabledOtlpRules has been set as istio-controlplane,k8s-node,oap,vm,mysql,postgresql
```
I will test in linux later, may be some problems with my computer.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json
Comment (yswdqz) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:362]: I get it.

**Comment by kezhenxu94** [comment]
@yswdqz please keep all tests in OC format for now, I'll test all features in showcase and update

**Comment by wu-sheng** [comment]
> Yes, I have resolved the conflicts and keep all things in OC format?

Yes, please. I don't think you should be blocked by a new protocol integration

**Comment by yswdqz** [comment]
> > Yes, I have resolved the conflicts and keep all things in OC format?
> 
> Yes, please. I don't think you should be blocked by a new protocol integration

OK, I have pushed them.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:300]:
```suggestion
                  "title":"Inserted Rows Trend (rows per second)"
```
I found your expression is:
```yaml
  - name: inserted_rows_rate
    exp: pg_stat_database_tup_inserted.sum(['datname','server']).rate('PT1M')
```
`rate(Duration)` means: Calculates the per-second `average rate of increase` in the time range, it should be `rows per second`
If you want to calculate `rows per minute`, you should use:
`increase(Duration)`: Calculates the increase in the time range.

I think using `rate` here is correct, just need to fix the unit, and please check your previous `MySQL monitoring PR` and make these consistently.

Comment (yswdqz) [reply oap-server/server-starter/src/main/resources/ui-initialized-templates/postgresql/postgresql-service.json:300]: It's my fault, I'm going to calculate `per second`, I'll change it now.

**Review by wankai123 — APPROVED** [review]
Thanks, LGTM

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/Layer.java:134]: @Fine0830 Please add this layer available to the menu as root, and put PostgreSQL as a sub-menu in the Database side by side with MySQL, name `PostgreSQL`

**Pull request merged**
